app/services/moves/finder.rb

```diff
@@ -84,5 +85,13 @@ def apply_supplier_filters(scope)
 
       scope.served_by(filter_params[:supplier_id])
     end
+
+    def apply_allocation_relationship_filters(scope)
+      return scope unless filter_params.key?(:has_relationship_to_allocation)
+
+      scope = scope.where.not(allocation: nil) if filter_params[:has_relationship_to_allocation] == 'true'
```

Comment:
Could this be changed to use `allocation_id` rather than `allocation`? I'm thinking it might be a tad faster...